subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf

```diff
@@ -256,6 +256,33 @@ def validateInputParameters() {
         }
     }
 
+    // Check that unaligned reads are saved when using Kraken2
+    if (params.kraken_db) {
+        if (!params.save_unaligned) {
```

Comment:
I think, that logic should not require human interaction. Instead, the unmapped reads should be generated (but not published) if contaminant screening is desired. 

For the STAR align module, one would e.g. change the line 

`${params.save_unaligned ? '--outReadsUnmapped Fastx' : ''}` to something like
`${params.save_unaligned || params.contaminant_screening != 'off' ? '--outReadsUnmapped Fastx' : ''}`

This ensures that the unaligned reads are technically available for the contamination screening, but still retain the user choice if they want to publish them to their output directory or not.

Reply:
Resolved with [6dd7521](https://github.com/nf-core/rnaseq/pull/1351/commits/6dd752109e68ebb142bccc48a822593ad37743f4)